### PR TITLE
[ie/LCI] Fix extractor

### PR DESCRIPTION
**Pull request by ocococococ**
**IMPORTANT**: PRs without the template will be CLOSED

### LCI web site changed recently and LCI extractor could not work anymore

- So I extended the _VALID_URL regular expression to match the new additional level in URL
- I added code to parse the web page in order to get wat_id from nextjs data (without this, a wrong wat_id for an unrelated video is retrieved)
- I added a test

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/lci.py [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
-        wat_id = self._search_regex(r'watId["\']?\s*:\s*["\']?(\d+)', webpage, 'wat id')
-        return self.url_result('wat:' + wat_id, 'Wat', wat_id)
+
+        next_data = self._search_nextjs_data(webpage, video_id)
+
+        videos = traverse_obj(next_data, ('props', 'pageProps', 'page', 'tms', 'videos'))
+        videos_id = videos.keys()
+
+        wat_id = [video_id for video_id in videos_id if video_id.isnumeric()][0]
+        return self.url_result('wat:{}'.format(wat_id), 'Wat', wat_id)
```
Comment (seproDev) [inline yt_dlp/extractor/lci.py:50]: How about:
```suggestion
        wat_id = traverse_obj(next_data, (
            'props', 'pageProps', 'page', 'tms', 'videos', {dict.keys}, ..., {int_or_none}, any))

        return self.url_result(f'wat:{wat_id}', WatIE, wat_id)
```
Also do any pages exist with multiple videos, in which case we should return this as a playlist?

File: yt_dlp/extractor/lci.py [hunk 2]
```diff
@@ -1,28 +1,50 @@
 from .common import InfoExtractor
 
+from ..utils.traversal import traverse_obj
```
Comment (seproDev) [inline yt_dlp/extractor/lci.py:3]:
```suggestion
from .wat import WatIE
from ..utils import int_or_none
from ..utils.traversal import traverse_obj
```

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/lci.py [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
-            'duration': 33,
-        },
-        'params': {
-            'skip_download': True,
-        },
-    }, {
-        'url': 'https://www.lci.fr/politique/election-presidentielle-2022-second-tour-j-2-marine-le-pen-et-emmanuel-macron-en-interview-de-lci-vendredi-soir-2217486.html',
-        'only_matching': True,
-    }]
+    _VALID_URL = r'https?://(?:www\.)?(?:lci|tf1info)\.fr(?:/[^/]+)+/[\w-]+-(?P<id>\d+)\.html'
```
Comment (bashonly) [inline yt_dlp/extractor/lci.py:8]:
```suggestion
    _VALID_URL = r'https?://(?:www\.)?(?:lci|tf1info)\.fr/(?:[^/?#]+/)+[\w-]+-(?P<id>\d+)\.html'
```

File: yt_dlp/extractor/lci.py [hunk 4]
```diff
@@ -1,28 +1,53 @@
...
-        wat_id = self._search_regex(r'watId["\']?\s*:\s*["\']?(\d+)', webpage, 'wat id')
-        return self.url_result('wat:' + wat_id, 'Wat', wat_id)
+
+        next_data = self._search_nextjs_data(webpage, video_id)
+
+        wat_id = traverse_obj(next_data, (
+            'props', 'pageProps', 'page', 'tms', 'videos', {dict.keys}, ..., {int_or_none}, any))
+
+        if not wat_id:
+            raise ExtractorError('Could not find wat_id')
```
Comment (bashonly) [inline yt_dlp/extractor/lci.py:51]:
```suggestion
        next_data = self._search_nextjs_data(webpage, video_id)
        wat_id = traverse_obj(next_data, (
            'props', 'pageProps', 'page', 'tms', 'videos', {dict.keys}, ..., {int_or_none}, any))
        if wat_id is None:
            raise ExtractorError('Could not find wat_id')
```

File: yt_dlp/extractor/lci.py [hunk 5]
```diff
@@ -1,28 +1,53 @@
...
+
+        return self.url_result(f'wat:{wat_id}', WatIE, wat_id)
```
Comment (bashonly) [inline yt_dlp/extractor/lci.py:53]:
```suggestion
        return self.url_result(f'wat:{wat_id}', WatIE, str(wat_id))
```

**Review by bashonly — APPROVED** [review]

**Pull request merged**